### ISSUE TITLE
Large number of files take a long time load - turn off thumbnails?

### ISSUE THREAD

**Issue by MetadataDeluxe** [hvdwolf/jExifToolGUI#117]
I selected a folder of 320 JPGs and it took over 5 minutes to load. This is probably due to process to generate thumbnails, but it might be that metadata is also being read for each of the files. 

In order to work with large batches (especially useful for the Export/Import and ExifTool Command functions). Would turning off thumbnail generation speed things up? If so, it might be good to have a checkbox underneath the file selection button, so the user can switch views quickly (no thumbs to run a batch, then turn on thumbs to make further edits to smaller groups or single images).

![jExifTool_UI_003](https://user-images.githubusercontent.com/5862936/116796618-ad0a1b80-aa92-11eb-85a3-3752e2ed06e1.jpg)

Another idea is to provide three checkboxes to control thumbs, filename and details separately.

![jExifTool_UI_002](https://user-images.githubusercontent.com/5862936/116796636-d9259c80-aa92-11eb-8204-723199ef0928.jpg)

You'll see that, in order to keep piling on the requests, I added a directory path display above the file list. I think this would be helpful to remember your place when working on complex tasks across directories.

**Comment by MetadataDeluxe** [comment]
Another option would be to add a folder browser to the Export-Import and ExifTool Commands screens (similar to what you have in the Geotagging screen).

![jExifTool_UI_004](https://user-images.githubusercontent.com/5862936/116797706-7dacdc00-aa9d-11eb-87f1-ae5ee7aaaa14.jpg)

**Comment by hvdwolf** [comment]
Beware: Long answer!

These are actually multiple requests in one issue.
1. Option to turn off thumbnails
2. Option to select between thumbs, filename and file details separately
3. Display source folder of images above loaded files list
4. Add directory option (without loading images) to act upon

The design principle is indeed that you work on smaller sets of images, but of course: that is based on my way of working as that is how the app originated (as pyExifToolGUI at that time). If I go out, I make 30, 50, 200 photos. I scan them first and remove 20-80%. Those remaing images I work on. For my way of working I really need to see the image to know on what image(s) I work on.
Note also that modern cameras (after 2010 or so?), always embed a preview in their jpgs/tiffs/RAWs. This makes loading much faster as I do not have to do anything apart from retrieving a very small jpg.

**1.** I will have a look at that. As such simply not looking for an embedded preview or creating one should not be too difficult. 
- Another point here is that images from a camera (somewhere after 2010) always have an embedded preview in the jpg (or tiff or raw). I try to use that one and it speeds up loading images tremendously because I do not have to resize and scale. Your scanned images might not have this preview.
- Martin already asked for an option to cache thumbnails "somewhere", so that on re-read of the same images the process is faster. I already did a few tests with that, but it in case of embedded previews it hardly is and improvement. Only for (huge) tiffs and other images without embedded preview it really helps. This is exactly Martin's area and maybe yours as well.
- I currently have the option to select one-column/2-column (Preferences -> Look & Feel), but that still always loads the preview. I might add other options as well, as you requested, and put it on the main screen (also in combination with next point).
- I'm also looking at other options right now. Currently you have the tabular form (1 column or 2 columns). I designed the app to run on a 1200x800/13766x768 screen, but users working on images are mostly using high-res screens of at least 1920x1080, so I am thinking about 1600x900 or indeed 1920x1080 as "default template". And instead of using a tabular view for the images, I might switch to a grid view. Depending on the width the user drags his/her windows, the previews will be displayed in rows of 3, 4, 5, .. images.

**2**. I am currently working on search through metadata. Say you load 320 jpgs (some people do ;) ), you click the search button and can search for "something" in either the keys and tags. To do that I need to read all info when loading those images. So when not reading file details, I will disable the search button (after loading).

**3.** That is indeed a good one. It should be a scrollable field though as I sometimes have very long paths (/mnt/chromeos/removable/256GB/Afbeeldingen/panasonic_share/20210422/<images names>). 
Only issue is that you can specify on the command line options of several folders and several files in several paths (to allow users to select "related" images in several locations). Of course that will hardly happen but in that case I need to disable the path and put something in like "multiple folders" or nothing.

**4.**  I like that one. I also use it for renaming but had not thought of using it in the "Export/Import" tab or "Exiftool Commands" tab. It can be very handy in combination with "favorites" commands in "Exiftool Commands".

**Comment by hvdwolf** [comment]
And there is even another option which I'm looking at.
I am thinking of making the left panel (previews) also a tabbed panel: previews/folder view.
- previews: as is.
- folder view: just like a file manager displaying a folder with folder names. If you click one image you get a preview beneath the folder tree and the metadata info on the right. If you double-click the filename, it will open for full view (like now), and of course with the multi-select fielnames you can do the same things as now.

So many things, so little time.

**Comment by MetadataDeluxe** [comment]
Yes, lots of requests bundled here :) They are just ideas to throw into the pot, knowing there are many other priorities.

Definitely, when entering descriptive metadata, it really helps to see the image. The large batch processes I'm thinking of would be things like updating the creator's contact info in older files or adding a keyword (after thinking about it later) to a large job folder. These tasks could involve thousands of files. These are the kind of tasks ExifTool is perfect for, but many people cannot work with the command line exe, the need a GUI. 

1. 
- The directory of 320 JPGs I tested were all from a Nikon D5100 and I think they have embedded previews. My computer is pretty fast, but I wonder if there might be other factors, like a corrupt image, or an unsupported file type.
- An image cache might be the answer, but it might bring complications. It is worth investigating.
- Ultimately a grid view for previews would be best. It would make it easier to review a lot of files and select related images for description. e.g., all photos of the same person. Screen size is a consideration. I agree that most users will be using a large desktop monitor, but the ability to resize to fit a lower resolution laptop is necessary.

2.
- Good idea to disable the search button if the file details are not read.

3.
- "multiple folders" is a good idea for several folders and several files in several paths

4.
- The radio button to choose a folder has the advantage of skipping all file reading, preview and details, so there would be no waiting at all. This would be great for export-import where people often work with thousands of images (many people have told me this). For example, you might export a directory (including sub-directories) to CSV, make a lot of complex find-and-replace edits in a spreadsheet, then import back to the images.

5. making the left panel (previews) also a tabbed panel: previews/folder view
- This is a great idea - very fast navigation and helpful when working between directories.

**Comment by hvdwolf** [comment]
feature request 4:
Your description is nice and short (use selected .... or Use all images ..) and leaves much space in the ui. For the time being I did it like this to also explain it. I might switch back to your short description.
Still working on expToPDF. That works entirely different using an external java component.
To be tested (further) on Windows and entirely on MacOS.
![image](https://user-images.githubusercontent.com/4520361/117148887-f621de80-adb6-11eb-986e-79c7ee83aa5f.png)

**Comment by MetadataDeluxe** [comment]
Looks good. I think this will be very useful for a lot of people.

**Comment by MetadataDeluxe** [comment]
Good solution - there is plenty of room along the bottom for long paths.

**Comment by hvdwolf** [comment]
As mentioned earlier via mail (but also for those following these issues):

![image](https://user-images.githubusercontent.com/4520361/120624189-ba833e80-c460-11eb-838e-eb598a7df621.png)

It is now possible to:

- not load/create previews
-  not load metadata (for all loaded images). 
 
Currently in development tree.
